core/src/main/java/com/novoda/noplayer/listeners/PreparedListeners.java

```diff
@@ -10,6 +10,8 @@
 
     private final Set<Player.PreparedListener> listeners = new CopyOnWriteArraySet<>();
 
+    private boolean hasPrepared = false;
```

Comment:
We moved the state up a level because it made more sense here than in each of the individual listeners.